### ISSUE TITLE
Fix for missing module (transformer, bytes , wget) 

### ISSUE THREAD

**Issue by ax-anoop** [TheLastBen/fast-stable-diffusion#782]
Seems like the deps have been removed from github? My training was failing, added the following at the top

!pip install wget 
!pip install transformers
!pip install bytes
!pip install bitsandbytes

Training in process as I type.

**Comment by t2ac32** [comment]
just got the same problem a couple minutes ago where did you add the installs?

**Comment by ax-anoop** [comment]
Yeah, add it to the top cell. Let me know if it works.

**Comment by articulite** [comment]
> !pip install wget
> !pip install transformers
> !pip install bytes
> !pip install bitsandbytes

It OOMs for me

**Comment by gorborukov** [comment]
> > 
> 
> Yeah, add it to the top cell. Let me know if it works.
in my case it works

**Comment by ReEnMikki** [comment]
I just changed wget to gdown.

**Comment by ax-anoop** [comment]
Anyone fix the runtime web stuff ? 

`
Already up to date.
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
[<ipython-input-21-97fe3dc26cd8>](https://localhost:8080/#) in <module>
    137   srv= getoutput('cat /content/srvr.txt')
    138 
--> 139   for line in fileinput.input('/usr/local/lib/python3.8/dist-packages/gradio/blocks.py', inplace=True):
    140     if line.strip().startswith('self.server_name ='):
    141         line = f'            self.server_name = "{srv[8:]}"\n'

1 frames
[/usr/lib/python3.8/fileinput.py](https://localhost:8080/#) in _readline(self)
    340                     pass
    341                 # The next few lines may raise OSError
--> 342                 os.rename(self._filename, self._backupfilename)
    343                 self._file = open(self._backupfilename, self._mode)
    344                 try:

FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.8/dist-packages/gradio/blocks.py' -> '/usr/local/lib/python3.8/dist-packages/gradio/blocks.py.bak'
`
Gtting this issue.

**Comment by ax-anoop** [comment]
Did you run 
!pip install wget 
?

**Comment by ax-anoop** [comment]
To get the webui running: 

!pip install -r /content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements.txt

**Comment by ReEnMikki** [comment]
> !pip install -r /content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements.txt

ERROR: Could not open requirements file: [Errno 2] No such file or directory: '/content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements.txt'

**Comment by ax-anoop** [comment]
> 
Did you mount to another dir ? Check your folders to see where that file is

**Comment by ax-anoop** [comment]
> > Did you run !pip install wget ?
> 
> Using gdown, and pip install wget it ran, but it was unable to find the model i trained earlier. and then error when training new. ![image](https://user-images.githubusercontent.com/94848873/204968413-b4119f99-f0f2-4794-b9e7-26bc9d296fcd.png)

Looks like you're still missing transformers lib, did you run pip install transformers ?

**Comment by ax-anoop** [comment]
finally, everything working, trainig + webui :D

**Comment by t2ac32** [comment]
@ax-anoop perhaps you could upload the notebook to a colab for other to follow until repo gets fixed

**Comment by ReEnMikki** [comment]
> @ax-anoop perhaps you could upload the notebook to a colab for others to follow until repo gets fixed

This, I agree. I am trying to follow his solutions but somehow it either still doesn't work, or it works, but another problem arises (I also already issued)

**Comment by ReEnMikki** [comment]
> > > Did you run !pip install wget ?
> > 
> > 
> > Using gdown, and pip install wget it ran, but it was unable to find the model i trained earlier. and then error when training new. ![image](https://user-images.githubusercontent.com/94848873/204968413-b4119f99-f0f2-4794-b9e7-26bc9d296fcd.png)
> 
> Looks like you're still missing transformers lib, did you run pip install transformers ?

Yes I already added the 4 pip install lines you mentioned but this error still happens. I hope you share your colab

**Comment by ax-anoop** [comment]
I will upload the colab, I just need to fix it so it works in one go. I will do in 6/7 hours, it's 1am for me

**Comment by t2ac32** [comment]
> 

same for me i'll keep trying in the morning nite everyone :)

**Comment by ReEnMikki** [comment]
Gn guys it's noon for me so I'll keep experimenting ig

**Comment by ReEnMikki** [comment]
> I will upload the colab, I just need to fix it so it works in one go. I will do in 6/7 hours, it's 1am for me

I think I'm fine with the colab even if it doesn't work in one go too tbh

**Comment by abvgdeabvgde2** [comment]
```
!pip install wget
!pip install transformers
!pip install bytes
!pip install bitsandbytes
!pip install clip
!pip install open_clip_torch
!pip install -r /content/gdrive/MyDrive/sd/stable-diffusion-webui/requirements.txt
```
I can confirm, these imports solve the problem. Both training and webui are working. Tested on 1.5 version.

**Comment by TheLastBen** [comment]
it's all fixed now, just use the latest colab

**Comment by ReEnMikki** [comment]
> it's all fixed now, just use the latest colab

thanks man lemme try

**Comment by ax-anoop** [comment]
> it's all fixed now, just use the latest colab

Sweet!

**Comment by ax-anoop** [comment]
FATAL: this function is for sm80, but was built for sm750

Anyone getting this during training ? With the new notebook.

**Comment by TheLastBen** [comment]
it's fixed now, the A100 xformers work, update to the latest colab.

**Comment by ax-anoop** [comment]
Yup seems to be training, thanks a ton Ben!

**Comment by ax-anoop** [comment]
> it's fixed now, the A100 xformers work, update to the latest colab.

Hey Ben, was hoping to connect with you about a different project/opportunity! Followed you on twitter, can't message unless you follow back :) Would love to chat very breifly.

**Comment by TheLastBen** [comment]
sure, no problem
